server/controllers/finance/reports/income_expense_by_month/index.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * Income Expense Controller
+ *
+ * This controller is responsible for processing the income and expense report.
+ *
+ * @module finance/incomeExpense
+ *
+ * @requires lodash
+ * @requires lib/db
+ * @requires lib/ReportManager
+ * @requires lib/errors/BadRequest
+ * @requires lib/Tree
+ * @requires constrollers/fiscal
+ */
+
+
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const ReportManager = require('../../../../lib/ReportManager');
+const BadRequest = require('../../../../lib/errors/BadRequest');
+const Tree = require('../../../../lib/Tree');
+const Fiscal = require('../../fiscal');
+
+const TEMPLATE = './server/controllers/finance/reports/income_expense_by_month/report.handlebars';
+
+const INCOME_TYPE_ID = 4;
+const EXPENSE_TYPE_ID = 5;
+const TITLE_ID = 6;
+
+const DEFAULT_PARAMS = {
+  csvKey : 'rows',
+  filename : 'TREE.INCOME_EXPENSE',
+  orientation : 'landscape',
+  footerRight : '[page] / [toPage]',
+};
+
+// expose to the API
+exports.document = document;
+
+function document(req, res, next) {
+  let report;
+
+  const options = _.defaults(req.query, DEFAULT_PARAMS);
+
+  try {
+    report = new ReportManager(TEMPLATE, req.session, options);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  const { periodId, periodNumber, fiscalYearId } = options;
+  const data = {};
+
+  Promise.all([
+    getPeriodByNumberAndFiscalId(parseInt(periodNumber, 10) - 1, fiscalYearId),
+    getPeriodById(periodId),
+    getPeriodByNumberAndFiscalId(parseInt(periodNumber, 10) + 1, fiscalYearId),
```

Comment:
This should be the current period (Period N), the previous period (Period N -1), and the period before that one (Period N - 2).

Reply:
This means the report compares the last tree monts' balanace?

Reply:
Yes sir!